spec/concepts/daily_update/task/adapt_etablissement_spec.rb

```diff
@@ -3,30 +3,43 @@
 describe DailyUpdate::Task::AdaptEtablissement do
   subject { described_class.call result: etablissement_insee }
 
-  let(:fixture_path) { 'spec/fixtures/samples_insee/etablissement.json' }
   let(:etablissement_insee) do
     JSON.parse(
       File.read(fixture_path),
       symbolize_names: true
     )
   end
 
-  let(:keys_to_ignore) { rails_keys + geo_keys }
-  let(:rails_keys) { %i[id unite_purgee created_at updated_at unite_legale_id] }
-  let(:geo_keys) { %i[longitude latitude geo_score geo_type geo_adresse geo_id geo_ligne geo_l4 geo_l5] }
-
-  let(:expected_keys) do
-    Etablissement
-      .new
-      .attributes
-      .deep_symbolize_keys
-      .tap { |hash| keys_to_ignore.each { |k| hash.delete(k) } }
-      .keys
+  describe 'etablissement diffusable' do
+    let(:fixture_path) { 'spec/fixtures/samples_insee/etablissement.json' }
+    let(:keys_to_ignore) { rails_keys + geo_keys }
+    let(:rails_keys) { %i[id unite_purgee created_at updated_at unite_legale_id] }
+    let(:geo_keys) { %i[longitude latitude geo_score geo_type geo_adresse geo_id geo_ligne geo_l4 geo_l5] }
+
+    let(:expected_keys) do
+      Etablissement
+        .new
+        .attributes
+        .deep_symbolize_keys
+        .tap { |hash| keys_to_ignore.each { |k| hash.delete(k) } }
+        .keys
+    end
```

Comment:
Je trouve toujours que la perte en clarté au setup des tests ne l'emporte pas sur la nécessité de tout ceci, mais bref...

Mais maintenant je me demande aussi pourquoi il faut ignorer les infos relatives aux données géographique ?

Reply:
je vais mettre tous les champs dans une liste alors.

Les données géographique ça vient de data.cquest.org, donc dans les DailyUpdate on ne peut donc pas les avoir :)

Reply:
Ok je vois. En fait ce qui me gène dans cette manière de faire c'est qu'il y a du comportement qui est caché dans le setup des tests... Au lieu d'avoir différentes specs du genres : 
`it 'updates attributes fetched from INSEE API'`
`it 'doesn't update rails timestamps and id'` 
`it 'doesn't update geo data' ` 

là il faut deviner le comportement depuis le setup du seul test, et personnellement je ne trouve pas ça clair. Et j'aimerai pas avoir à investiguer et corriger le test si jamais il ne passe pas dans le futur :/

Reply:
C'est résolu avec une liste explicite des champs